packages/expo-cli/src/projects.ts

```diff
@@ -0,0 +1,33 @@
+import { ProjectPrivacy } from '@expo/config';
+import { ApiV2, User } from '@expo/xdl';
+
+interface ProjectData {
+  accountName: string;
+  projectName: string;
+  privacy?: ProjectPrivacy;
+}
+
+async function ensureProjectExistsAsync(
+  user: User,
+  { accountName, projectName, privacy }: ProjectData
+): Promise<string> {
+  const client = ApiV2.clientForUser(user);
+  try {
+    const experienceName = `@${accountName}/${projectName}`;
+    const [{ id }] = await client.getAsync('projects', { experienceName });
+    return id;
+  } catch (err) {
+    if (err.code !== 'EXPERIENCE_NOT_FOUND') {
+      throw err;
+    }
+  }
+
+  const { id } = await client.postAsync('projects', {
```

Comment:
do you think it's worth prompting users before creating the project to verify that they would like to create a project with this accountName/projectName?

Reply:
Hmm, I'm not against it but this prompt seems irrelevant. It'd be another decision for the user to make and I think we should prompt users as rarely as possible. However, I think it can be helpful to add a log saying that `This project .... hasn't been registered on Expo servers yet. Creating a record for it.` or sth similar.

Reply:
@brentvatne check it out now :)

Reply:
that looks good to me!